src/query/storage/promremote/storage_test.go

```diff
@@ -0,0 +1,253 @@
+	promShortRetention, closeFn1 := promremotetest.NewServer(t)
+	defer closeFn1()
+	promMediumRetention, closeFn2 := promremotetest.NewServer(t)
+	defer closeFn2()
+	promLongRetention, closeFn3 := promremotetest.NewServer(t)
+	defer closeFn3()
+	promLongRetention2, closeFn4 := promremotetest.NewServer(t)
+	defer closeFn4()
+	reset := func() {
+		promShortRetention.Reset()
+		promMediumRetention.Reset()
+		promLongRetention.Reset()
+	}
+
+	promStorage, err := NewStorage(Options{
+		endpoints: []EndpointOptions{
+			{
+				address:    promShortRetention.WriteAddr(),
+				retention:  120 * time.Hour,
+				resolution: 15 * time.Second,
+			},
+			{
+				address:    promMediumRetention.WriteAddr(),
+				retention:  720 * time.Hour,
+				resolution: 5 * time.Minute,
+			},
+			{
+				address:    promLongRetention.WriteAddr(),
+				retention:  8760 * time.Hour,
+				resolution: 10 * time.Minute,
+			},
+			{
+				address:    promLongRetention2.WriteAddr(),
+				retention:  8760 * time.Hour,
+				resolution: 10 * time.Minute,
+			},
+		},
+		scope: tally.NoopScope,
+	})
+	require.NoError(t, err)
+	defer closeStorage(t, promStorage)
+
+	sendWrite := func(attr storagemetadata.Attributes) error {
+		//nolint: gosec
+		datapoint := ts.Datapoint{Value: rand.Float64(), Timestamp: xtime.Now()}
+		wq, err := storage.NewWriteQuery(storage.WriteQueryOptions{
+			Tags: models.Tags{
+				Opts: models.NewTagOptions(),
+				Tags: []models.Tag{{
+					Name:  []byte("test_tag_name"),
+					Value: []byte("test_tag_value"),
+				}},
+			},
+			Datapoints: ts.Datapoints{datapoint},
+			Unit:       xtime.Millisecond,
+			Attributes: attr,
+		})
+		require.NoError(t, err)
+		return promStorage.Write(context.TODO(), wq)
+	}
+
+	t.Run("send short retention write", func(t *testing.T) {
+		reset()
+		err := sendWrite(storagemetadata.Attributes{
+			Retention:  120 * time.Hour,
+			Resolution: 15 * time.Second,
+		})
+		require.NoError(t, err)
+		assert.NotNil(t, promShortRetention.GetLastRequest())
+		assert.Nil(t, promMediumRetention.GetLastRequest())
+		assert.Nil(t, promLongRetention.GetLastRequest())
+	})
+
+	t.Run("send medium retention write", func(t *testing.T) {
+		reset()
+		err := sendWrite(storagemetadata.Attributes{
+			Resolution: 5 * time.Minute,
+			Retention:  720 * time.Hour,
+		})
+		require.NoError(t, err)
+		assert.Nil(t, promShortRetention.GetLastRequest())
+		assert.NotNil(t, promMediumRetention.GetLastRequest())
+		assert.Nil(t, promLongRetention.GetLastRequest())
+	})
+
+	t.Run("send long retention write", func(t *testing.T) {
+		reset()
+		err := sendWrite(storagemetadata.Attributes{
+			MetricsType: storagemetadata.AggregatedMetricsType,
+			// Should be ignored when type is unagg
+			Resolution: 10 * time.Minute,
+			Retention:  8760 * time.Hour,
+		})
+		require.NoError(t, err)
+		assert.Nil(t, promShortRetention.GetLastRequest())
+		assert.Nil(t, promMediumRetention.GetLastRequest())
+		assert.NotNil(t, promLongRetention.GetLastRequest())
+	})
```

Comment:
I think this test case can be dropped in favour of `send write to multiple instances configured with same retention`.
I don't see anything specific about this third retention (vs. the other two).